torch/_refs/__init__.py

```diff
@@ -3473,6 +3474,26 @@ def diag(
         return torch.diagonal_copy(self, offset)
 
 
+@register_decomposition(torch.ops.aten.diagonal_scatter)
+@out_wrapper()
+def diagonal_scatter(
+    input: TensorLikeType,
+    src: TensorLikeType,
+    offset: int = 0,
+    dim1: int = 0,
+    dim2: int = 1,
+) -> TensorLikeType:
+    out = input.clone()
```

Comment:
Why the clone here?

Reply:
Oh, you're copying to `diag`. Maybe we should have required named arguments for `copy_to` so we'd never forget which was the source and which the destination.